### ISSUE TITLE
Review public API of web2

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2084]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#2084|
|Type|enhancement|
|Created|2006-09-13 00:43:10Z|

The goal of this review is to make sure that the API is something we are all willing to declare stable, and also to figure out (and document) which parts are declared stable.

Since some of the web2 code is pre-UQDS, we should not turn this into a comprehensive code review.  The review should take place on the pydoctor-generated API documentation.

The output of this review should be:

 * a list of suggested changes to the APIs as presented
 * a list of currently public classes, functions, and modules which should be made private (e.g. prefixed with "_")
 * recommendations for any bad or dead code which should be removed entirely
 * a list of work-in-progress APIs to mark as "unstable"

This bug should be closed only once all of these recommendations have been addressed - please create additional tickets for required development work and link them from this ticket.

<details><summary>Searchable metadata</summary>

```
trac-id__2084 2084
type__enhancement enhancement
reporter__glyph glyph
priority__high high
milestone__Web2_Gold_Master Web2-Gold-Master
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__web2 web2
keywords__ 
time__1158108190000000 1158108190000000
changetime__1222195439000000 1222195439000000
version__None None
owner__ 
cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set status to `closed`|
|-|-|

web2 will remain as it is.  Features will be ported to `twisted.web` individually (or in small groups, where that makes sense).  As each feature is ported, the API can be reviewed in the same manner as is usual for code reviews of new features.  So, nothing to do here.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
